Dnn.AdminExperience/ClientSide/Servers.Web/src/components/Tabs/SmtpServer.jsx

```diff
@@ -11,278 +11,336 @@ import SmtpServerTabActions from "../../actions/smtpServerTab";
+      smtpAuthentication: smtpSettings.smtpAuthentication,
+      smtpUsername: smtpSettings.smtpUserName,
+      smtpPassword: smtpSettings.smtpPassword,
+      enableSmtpSsl: smtpSettings.enableSmtpSsl,
+    };
+    props.onSendTestEmail(sendEmailRequest);
+  }
 
-        if (this.areThereValidationError()) {
-            return;
-        } 
+  getSmtpServerOptions() {
+    return [
+      {
+        label: localization.get("GlobalSmtpHostSetting"),
+        value: "h",
+      },
+      {
+        label: localization
+          .get("SiteSmtpHostSetting")
+          .replace("{0}", this.props.smtpServerInfo.portalName || ""),
+        value: "p",
+      },
+    ];
+  }
 
-        let smtpSettings = {};
-        if (props.smtpServerInfo.smtpServerMode === "h" && utils.isHostUser()) {
-            smtpSettings = props.smtpServerInfo.host;
-        }
-        if (props.smtpServerInfo.smtpServerMode === "p") {
-            smtpSettings = props.smtpServerInfo.site;
-        }
-        
-        const sendEmailRequest = {
-            smtpServerMode: props.smtpServerInfo.smtpServerMode,
-            smtpServer: smtpSettings.smtpServer,
-            smtpAuthentication: smtpSettings.smtpAuthentication,
-            smtpUsername: smtpSettings.smtpUserName,
-            smtpPassword: smtpSettings.smtpPassword,
-            enableSmtpSsl: smtpSettings.enableSmtpSsl
-        };
-        props.onSendTestEmail(sendEmailRequest);    
-    }   
+  getSmtpAuthenticationOptions() {
+    return [
+      {
+        label: localization.get("SMTPAnonymous"),
+        value: "0",
+      },
+      {
+        label: localization.get("SMTPBasic"),
+        value: "1",
+      },
+      {
+        label: localization.get("SMTPNTLM"),
+        value: "2",
+      },
+    ];
+  }
 
-    getSmtpServerOptions() {
-        return [{
-            label: localization.get("GlobalSmtpHostSetting"),
-            value: "h"
-        },
-        {
-            label: localization.get("SiteSmtpHostSetting").replace("{0}", this.props.smtpServerInfo.portalName || ""),
-            value: "p"
-        }
-        ];
-    }
+  render() {
+    const { props } = this;
+    const areGlobalSettings = props.smtpServerInfo.smtpServerMode === "h";
+    const selectedSmtpSettings =
+      (areGlobalSettings
+        ? props.smtpServerInfo.host
+        : props.smtpServerInfo.site) || {};
+    const credentialVisible = selectedSmtpSettings.smtpAuthentication === "1";
+    const smtpSettingsVisible = utils.isHostUser() || !areGlobalSettings;
 
-    getSmtpAuthenticationOptions() {
-        return [{
-            label: localization.get("SMTPAnonymous"),
-            value: "0"
-        },
-        {
-            label: localization.get("SMTPBasic"),
-            value: "1"
-        },
-        {
-            label: localization.get("SMTPNTLM"),
-            value: "2"
-        }
-        ];
+    if (props.smtpServerInfo.mailProvider !== "CoreMailProvider") {
```

Comment:
Rather than checking the name of the provider, can we check for an attribute on the provider?  That would allow some non-core providers to continue to re-use the SMTP configuration if it works for them.

So the config would look perhaps like this:
```xml
    <mail defaultProvider="SendGridMailProvider">
      <providers>
        <clear />
        <add name="MailKitMailProvider" type="DotNetNuke.Services.Mail.MailKitMailProvider, DotNetNuke" hideSmtpConfig="false" />
        <add name="SendGridMailProvider" type="DotNetNuke.Services.Mail.SendGridMailProvider, SomeThirdParty.SendGridMailProvider" hideSmtpConfig="true" />
      </providers>
    </mail>
```

Reply:
I briefly thought about that but decided not to. I'm not convinced it's worth it, given that every provider will have its own quirks and therefore extra settings (and then you'd get spread out settings). I'll let @mitchsellers chime in on that.

Reply:
OK, I've added an attribute for this.